subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/MavenMetadataLoader.java

```diff
@@ -53,8 +59,14 @@ public MavenMetadata load(URI metadataLocation) throws ResourceException {
         return metadata;
     }
 
-    private void parseMavenMetadataInfo(final URI metadataLocation, final MavenMetadata metadata) {
-        ExternalResource resource = repository.getResource(metadataLocation, true);
+    private void parseMavenMetadataInfo(final URI metadataLocation, final MavenMetadata metadata) throws IOException {
+        ExternalResource resource = cacheAwareExternalResourceAccessor.getResource(metadataLocation, new CacheAwareExternalResourceAccessor.ResourceFileStore() {
```

Comment:
I think the problem with uncached resources is that we were making multiple requests for the same resource in the _same build execution_: introducing caching with a zero timeout was a good way to fix this. But the focus of resource caching was supposed to be resources accessed by `ComponentMetadataSupplier`, not `maven-metadata.xml`.

The plan for caching changed during the implementation of this story, which might explain why we included `maven-metadata.xml` in the scope.